src/components/markdown-renderer/markdown-it-configurator.ts

```diff
@@ -0,0 +1,181 @@
+    langPrefix: '',
+    typographer: true
+  })
+
+  md.use(emoji, {
+    defs: combinedEmojiData
+  })
+  md.use(abbreviation)
+  md.use(definitionList)
+  md.use(subscript)
+  md.use(superscript)
+  md.use(inserted)
+  md.use(marked)
+  md.use(footnote)
+  md.use(imsize)
+
+  return md
+}
+
+const fullMarkdownConfigurator = (
+  useFrontmatter: boolean,
+  onYamlError: (error: boolean) => void,
+  onRawMeta: (rawMeta: RawYAMLMetadata) => void,
+  onToc: (toc: TocAst) => void,
+  onLineMarkers: (lineMarkers: LineMarkers[]) => void
+): MarkdownIt => {
+  const md = basicMarkdownConfigurator()
+
+  const plantumlServer = store.getState().config.plantumlServer
+
+  md.use(markdownItTaskLists, { lineNumber: true })
+  if (plantumlServer) {
+    md.use(plantuml, {
+      openMarker: '```plantuml',
+      closeMarker: '```',
+      server: plantumlServer
+    })
+  } else {
+    md.use(plantumlError)
+  }
+
+  if (useFrontmatter) {
+    md.use(frontmatter, (rawMeta: string) => {
+      try {
+        const meta: RawYAMLMetadata = yaml.safeLoad(rawMeta) as RawYAMLMetadata
+        onYamlError(false)
+        onRawMeta(meta)
+      } catch (e) {
+        console.error(e)
+        onYamlError(true)
+        onRawMeta({} as RawYAMLMetadata)
+      }
+    })
+  }
+  // noinspection CheckTagEmptyBody
+  md.use(anchor, {
+    permalink: true,
+    permalinkBefore: true,
+    permalinkClass: 'heading-anchor text-dark',
+    permalinkSymbol: '<i class="fa fa-link"></i>'
+  })
+  md.use(mathJax({
+    beforeMath: '<app-katex>',
+    afterMath: '</app-katex>',
+    beforeInlineMath: '<app-katex inline>',
+    afterInlineMath: '</app-katex>',
+    beforeDisplayMath: '<app-katex>',
+    afterDisplayMath: '</app-katex>'
+  }))
+  md.use(markdownItRegex, replaceLegacyYoutubeShortCode)
+  md.use(markdownItRegex, replaceLegacyVimeoShortCode)
+  md.use(markdownItRegex, replaceLegacyGistShortCode)
+  md.use(markdownItRegex, replaceLegacySlideshareShortCode)
+  md.use(markdownItRegex, replaceLegacySpeakerdeckShortCode)
+  md.use(markdownItRegex, replacePdfShortCode)
+  md.use(markdownItRegex, replaceAsciinemaLink)
+  md.use(markdownItRegex, replaceYouTubeLink)
+  md.use(markdownItRegex, replaceVimeoLink)
+  md.use(markdownItRegex, replaceGistLink)
+  md.use(highlightedCode)
+  md.use(markdownItRegex, replaceQuoteExtraAuthor)
+  md.use(markdownItRegex, replaceQuoteExtraColor)
+  md.use(markdownItRegex, replaceQuoteExtraTime)
+  md.use(toc, {
+    placeholder: '(\\[TOC\\]|\\[toc\\])',
+    listType: 'ul',
+    level: [1, 2, 3],
+    callback: (code: string, ast: TocAst): void => {
+      onToc(ast)
+    },
+    slugify: slugify
+  })
+  validAlertLevels.forEach(level => {
+    md.use(markdownItContainer, level, { render: createRenderContainer(level) })
+  })
+  md.use(lineNumberMarker(), {
+    postLineMarkers: (lineMarkers) => {
+      onLineMarkers(lineMarkers)
+    }
+  })
+  return md
+}
+
+export const configureLinkfyExtra = (md: MarkdownIt): MarkdownIt => {
```

Comment:
Maybe rename this to something like postMarkdownConfigurator or similar? It does hold more than only the linkify-extra module.

Reply:
How about a class structure for this? MarkdownConfigurator as an abstract class. BasicMarkdownConfigurator extends this. FullMarkdownConfigurator extends this.

Reply:
This could make this hole file and the flow a bit cleaner